include/deal.II/base/polynomials_hermite.h

```diff
@@ -0,0 +1,158 @@
+
+
+
+#ifndef dealii_hermite_polynomials
+#define dealii_hermite_polynomials
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/subscriptor.h>
+
+#include <deal.II/lac/full_matrix.h>
+
+#include <algorithm>
+#include <exception>
+#include <memory>
+#include <vector>
+
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * @addtogroup Polynomials
+ * @{
+ */
+
+namespace Polynomials
+{
+  /**
+   * This class implements Hermite interpolation polynomials (see 
+   * @cite CiarletRiavart1972interpolation) enforcing the maximum 
+   * posible level of regularity $r$ in the FEM basis given a 
+   * polynomial degree of $2r+1$. The polynomials all represent 
+   * either a non-zero shape value or derivative at $y=0$ and $y=1$ 
+   * on the reference interval $y \in [0,1]$.
+   *
+   * Indices $j = 0, 1, \dots, r$ refer to polynomials corresponding
+   * to a non-zero derivative (or shape value for $j=0$) of
+   * order $j$ at $x=0$, and indices $j = r+1, r+2, \dots, 2r+1$
+   * refer to polynomials with a non-zero derivative of order
+   * $j-(r+1)$ (or value for $j=r+1$) at $x=1$. The basis is rescaled
+   * so that a function corresponding to a non-zero $j^{th}$
+   * derivative has derivative value $j! 4^{j}$ at the corresponding
+   * node.
+   */
+  class HermitePoly : public Polynomial<double>
+  {
+  public:
+    /**
+     * Constructor for an individual Hermite polynomial. We write f_{j}
+     * for a polynomial that has a non-zero $j^{th}$ derivative at $y=0$
+     * and $g_{j}$ for a polynomial with a non-zero $j^{th}$ derivative
+     * at $y=1$, meaning $f_{j}$ will have @p index $=j$ and $g_{j}$ will
+     * have @p index $= j + \mathtt{regularity} + 1$. The resulting
+     * polynomials will be degree $2\times \mathtt{regularity} +1$
+     * and obey the following conditions:
+     *
+     * @f{align*}{
+     * &\begin{matrix}
+     *   \left. \frac{d^{i}}{dy^{i}} f_{j}(y) \right\vert_{y=0}
+     *          = i! 4^{i} \delta_{i, j}, \hfill
+     *          &\qquad \hfill 0 \leq i \leq \mathtt{regularity}, \\
+     *   \left. \frac{d^{i}}{dy^{i}} f_{j}(y) \right\vert_{y=1}
+     *          = 0, \hfill &\qquad \hfill 0 \leq i \leq \mathtt{regularity},
+     * \end{matrix} \qquad 0 \leq j \leq \mathtt{regularity}, \\
+     * &\begin{matrix}
+     *  \left. \frac{d^{i}}{dy^{i}} g_{j}(y) \right\vert_{y=0}
+     *          = 0, \hfill &\qquad \hfill 0 \leq i \leq \mathtt{regularity}, \\
+     * \left. \frac{d^{i}}{dy^{i}} g_{j}(y) \right\vert_{y=1}
+     *          = i! 4^{i} \delta_{i, j}, \hfill
+     *          &\qquad \hfill 0 \leq i \leq \mathtt{regularity},
+     * \end{matrix} \qquad 0 \leq j \leq \mathtt{regularity},
+     * @f}
+     *
+     * where $\delta_{i,j}$ is equal to $1$ whenever $i=j$,
+     * and equal to $0$ otherwise. These polynomials have explicit
+     * formulas given by
```

Comment:
The formula and the text that precedes and follows it form a sentence. An empty line denotes the end of a paragraph, which shouldn't interrupt the sentence -- so let's remove the empty lines before and after the formula:
```suggestion
     * and obey the following conditions:
     * @f{align*}{
     * &\begin{matrix}
     *   \left. \frac{d^{i}}{dy^{i}} f_{j}(y) \right\vert_{y=0}
     *          = i! 4^{i} \delta_{i, j}, \hfill
     *          &\qquad \hfill 0 \leq i \leq \mathtt{regularity}, \\
     *   \left. \frac{d^{i}}{dy^{i}} f_{j}(y) \right\vert_{y=1}
     *          = 0, \hfill &\qquad \hfill 0 \leq i \leq \mathtt{regularity},
     * \end{matrix} \qquad 0 \leq j \leq \mathtt{regularity}, \\
     * &\begin{matrix}
     *  \left. \frac{d^{i}}{dy^{i}} g_{j}(y) \right\vert_{y=0}
     *          = 0, \hfill &\qquad \hfill 0 \leq i \leq \mathtt{regularity}, \\
     * \left. \frac{d^{i}}{dy^{i}} g_{j}(y) \right\vert_{y=1}
     *          = i! 4^{i} \delta_{i, j}, \hfill
     *          &\qquad \hfill 0 \leq i \leq \mathtt{regularity},
     * \end{matrix} \qquad 0 \leq j \leq \mathtt{regularity},
     * @f}
     * where $\delta_{i,j}$ is equal to $1$ whenever $i=j$,
     * and equal to $0$ otherwise. These polynomials have explicit
     * formulas given by
```